styles/sass/layouts.scss

```diff
@@ -1548,4 +1548,15 @@ body.prepare-print {
             border-bottom: 0;
         }
     }
-}
\ No newline at end of file
+}
+
+.loading-overlay-for-publish {
```

Comment:
Try making things reusable and avoid unnecessary coupling. In this case, it's just a button that displays a spinner when in loading state. It isn't related to publishing except that you are using it there. Including publishing in the name makes it seem that it only works there, which shouldn't be the case.